java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java

```diff
@@ -962,7 +963,10 @@ public void setConvertVectorElement(int elementNum) throws IOException {
         seconds += 1;
         nanos = 1_000_000_000 - nanos;
       }
-      HiveDecimal value = HiveDecimal.create(String.format("%d.%09d", seconds, nanos));
+      BigDecimal secondsBd = new BigDecimal(seconds);
+      BigDecimal nanosBd = new BigDecimal(nanos).movePointLeft(9);
+      BigDecimal resultBd = (seconds >= 0L) ? secondsBd.add(nanosBd) : secondsBd.subtract(nanosBd);
```

Comment:
Do you have a test coverage for this negative second case? If we don't have, could you make one please?

Reply:
If I break this code intentionally (add instead of subtract), there are failures in:

```
[ERROR] Failures: 
[ERROR]   TestProlepticConversions.testSchemaEvolutionTimestamp:375 row 0 ==> expected: <-6212894324588> but was: <-6212894324412>
[ERROR]   TestProlepticConversions.testSchemaEvolutionTimestamp:375 row 0 ==> expected: <-6212877044588> but was: <-6212877044412>
[ERROR]   TestProlepticConversions.testSchemaEvolutionTimestamp:375 row 0 ==> expected: <-6212894324588> but was: <-6212894324412>
[ERROR]   TestProlepticConversions.testSchemaEvolutionTimestamp:375 row 0 ==> expected: <-6212877044588> but was: <-6212877044412>
```

Reply:
Thank you for checking. Got it. That looks like an enough test coverage.